src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UITests/Presentation/SampleChooserViewModel.cs

```diff
@@ -885,21 +885,25 @@ public async Task<object> UpdateContent(CancellationToken ct, SampleChooserConte
 
 			FrameworkElement container = null;
 
-			var frameRequested =
-				newContent.UsesFrame &&
-				typeof(Page).IsAssignableFrom(newContent.ControlType);
-			if (frameRequested)
+			if (newContent.UsesFrame &&
+				typeof(Page).IsAssignableFrom(newContent.ControlType))
 			{
-				var frame = new Frame();
-				frame.Navigate(newContent.ControlType);
-				container = frame;
+				// Creating a new Frame with dynamically instantiated content based on newContent.ControlType
+				// and encapsulating it within a SampleControl along with a description
+				var frame = new Frame { Content = Activator.CreateInstance(newContent.ControlType) };
+				container = new Uno.UI.Samples.Controls.SampleControl
+				{
+					Content = frame,
+					SampleDescription = newContent.Description
+				};
```

Comment:
I think this will make tests behave differently on iOS, because the top level control will not be a `Frame`.

Reply:
Right, instead it should be the other way around ->

Uno.UI.Samples.Controls.SampleControl -> Frame -> sample control type